bin_sort/BinSort.java

```diff
@@ -0,0 +1,75 @@
+import java.util.Random;
+
+class Node {
+	public double data;
+	public Node next;
+
+	public Node(double data) {
+		this.data = data;
+		next = null;
+	}
+}
+
+// Time Complexity O(n)-> avg case and O(n^2)-> Worst Case
+// constraints:- 0<=A[i]<1
+public class BinSort {
+	public static void binSort(Node[] b, double[] a) {
+		for (int i = 0; i < a.length; i++) {
+			int ins = (int) (a[i] * 10);

```

Comment:
@mohitkyadav `Math.floor()` with round 0.123 to 0, but in his case I think he wants it 1. Similarly 0.232 to 2. etc